grade-management-new/GradeManagement.Services/Services/SubjectService.cs

```diff
@@ -0,0 +1,80 @@
+using AutoMapper;
+using GradeManagement.Data.Data;
+using GradeManagement.Data.Models;
+using GradeManagement.Shared.DTOs;
+using Microsoft.EntityFrameworkCore;
+using Task = System.Threading.Tasks.Task;
+
+namespace GradeManagement.Services.Services;
+
+public class SubjectService
+{
+    private readonly GradeManagementDbContext _gradeManagementDbContext;
+    private readonly IMapper _mapper;
+
+    public SubjectService(GradeManagementDbContext gradeManagementDbContext, IMapper mapper)
+    {
+        _gradeManagementDbContext = gradeManagementDbContext;
+        _mapper = mapper;
+    }
+
+    public async Task<IEnumerable<SubjectDTO>> GetAllSubjects()
+    {
+        List<Subject> allSubjects = await _gradeManagementDbContext.Subject.OrderByDescending(s => s.Id).ToListAsync();
+        var subjectDtos = _mapper.Map<List<SubjectDTO>>(allSubjects);
+        return subjectDtos;
+    }
+
+    public async Task<SubjectDTO> GetSubjectById(long id)
+    {
+        Subject? subject = await _gradeManagementDbContext.Subject.SingleOrDefaultAsync(s => s.Id == id);
+        if (subject == null) throw new NullReferenceException("No subject could be found with the given Id!");
```

Comment:
Használhatjuk a ?? operátort a throw kulcsszóval. Sőt húzzuk be az én céges nugetemet, amiben ilyen segédfüggvények is vannak: https://github.com/ZenitechSoftware/AutSoftCore/blob/master/src/AutSoft.Linq/Queryable/FirstAndSingleExtensions.cs

Reply:
A NullRefException amúgy sem túl jó, nem elég specifikus, ha megnézed a segédfüggvényemet, az egy EntityNotFound-ot dob (használjuk a NuGet-ben lévőt).